liblangutil/Scanner.cpp

```diff
@@ -967,7 +967,17 @@ tuple<Token, unsigned, unsigned> Scanner::scanIdentifierOrKeyword()
 	while (isIdentifierPart(m_char) || (m_char == '.' && m_kind == ScannerKind::Yul))
 		addLiteralCharAndAdvance();
 	literal.complete();
-	return TokenTraits::fromIdentifierOrKeyword(m_tokens[NextNext].literal);
+	auto const token = TokenTraits::fromIdentifierOrKeyword(m_tokens[NextNext].literal);
+	if (m_kind == ScannerKind::Yul)
+	{
+		// Turn Solidity identifier into a Yul keyword
+		if (m_tokens[NextNext].literal == "leave")
```

Comment:
What about `leave.x`?

Reply:
What should it do in that case? `Token::Illegal`?

Reply:
Now that we disallow consecutive dots, we should probably move the processing of dot-separated identifers to the parser.

Reply:
Do you mean `leave.x` would turn into `Identifier Period Identifier` ?

Reply:
Not that you are currently running short on PRs... ;)

Reply:
I still prefer a separate PR with the hope this can be merged quicker. I think it is a self contained change.

Reply:
I think we should introduce a new token for identifiers with dots: `Token::Path`. This way we can be clear in the parser at which parts are dotted identifiers accepted and currently we do not care.